android/java/org/chromium/chrome/browser/playlist/hls_content/HlsServiceImpl.java

```diff
@@ -212,4 +221,12 @@ private void initPlaylistService() {
         mPlaylistService =
                 PlaylistServiceFactoryAndroid.getInstance().getPlaylistService(HlsServiceImpl.this);
     }
+
+    @Override
+    public void onDestroy() {
+        if (mPlaylistService != null) {
+            mPlaylistService.close();
```

Comment:
why do we need to call `close` when the object is getting destroyed anyway? That should happen automatically

Reply:
we have to explicitly close the connection as per the below warning/error, 
`java.lang.IllegalStateException: Warning: Router objects should be explicitly closed when no longer required otherwise you may leak handles.
                                                                                                    	at Hm.finalize(chromium-MonochromePublic.apk-canary-426400020:25)
                                                                                                    	at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:250)
                                                                                                    	at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:237)
                                                                                                    	at java.lang.Daemons$Daemon.run(Daemons.java:103)
                                                                                                    	at java.lang.Thread.run(Thread.java:764)
                                                                                                    Caused by: java.lang.Exception: AutocloseableRouter allocated at:
                                                                                                    	at Hm.<init>(chromium-MonochromePublic.apk-canary-426400020:16)
                                                                                                    	at jf1.a(chromium-MonochromePublic.apk-canary-426400020:12)
                                                                                                    	at Bm0.t(chromium-MonochromePublic.apk-canary-426400020:23)
                                                                                                    	at Ha.d(chromium-MonochromePublic.apk-canary-426400020:31)
                                                                                                    	at Ja.n(chromium-MonochromePublic.apk-canary-426400020:36)
                                                                                                    	at nF2.n(chromium-MonochromePublic.apk-canary-426400020:80)
                                                                                                    	at m90.a(chromium-MonochromePublic.apk-canary-426400020:45)
                                                                                                    	at l90.a(chromium-MonochromePublic.apk-canary-426400020:12)
                                                                                                    	at org.chromium.mojo.system.impl.WatcherImpl.onHandleReady(chromium-MonochromePublic.apk-canary-426400020:3)
                                                                                                    	at android.os.MessageQueue.nativePollOnce(Native Method)
                                                                                                    	at android.os.MessageQueue.next(MessageQueue.java:326)
                                                                                                    	at android.os.Looper.loop(Looper.java:181)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7050)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)`